packages/jellyfish-api-core/__tests__/category/loan/getLoanInfo.test.ts

```diff
@@ -0,0 +1,564 @@
+      fixedIntervalPriceId: 'TSLA/USD'
+    })
+    await testing.container.generate(1)
+
+    await testing.rpc.loan.createLoanScheme({
+      minColRatio: 150,
+      interestRate: new BigNumber(2),
+      id: 'scheme1'
+    })
+    await testing.container.generate(1)
+
+    const vault1Id = await testing.rpc.loan.createVault({
+      ownerAddress: await testing.generateAddress()
+    })
+    await testing.container.generate(1)
+
+    const tempAddress = await testing.generateAddress()
+    await testing.token.dfi({ amount: 10000, address: tempAddress })
+    await testing.container.generate(1)
+    await testing.rpc.loan.depositToVault({
+      vaultId: vault1Id,
+      from: tempAddress,
+      amount: '10000@DFI'
+    })
+    await testing.container.generate(1)
+
+    const extendedStartingData: GetLoanInfoResult = {
+      ...startingData,
+      currentPriceBlock: new BigNumber(108),
+      nextPriceBlock: new BigNumber(114),
+      defaults: {
+        ...startingData.defaults,
+        scheme: 'scheme1'
+      },
+      totals: {
+        ...startingData.totals,
+        schemes: new BigNumber(1),
+        collateralTokens: new BigNumber(1),
+        collateralValue: new BigNumber(10000),
+        openVaults: new BigNumber(1),
+        loanTokens: new BigNumber(1)
+      }
+    }
+
+    { // before
+      const data = await testing.rpc.loan.getLoanInfo()
+      expect(data).toStrictEqual(extendedStartingData)
+    }
+
+    await testing.rpc.loan.takeLoan({
+      vaultId: vault1Id,
+      amounts: '2@TSLA'
+    })
+    await testing.container.generate(1)
+
+    { // After loan taken
+      const data = await testing.rpc.loan.getLoanInfo()
+      expect(data).toStrictEqual({
+        ...extendedStartingData,
+        totals: {
+          ...extendedStartingData.totals,
+          loanValue: expect.any(BigNumber)
+        }
+      })
+      // loan value will include some interest
+      expect(data.totals.loanValue.gt(10)).toBeTruthy() // 2 * 5
+      expect(data.totals.loanValue.lt(10.0001)).toBeTruthy()
+    }
+
+    await testing.rpc.loan.takeLoan({
+      vaultId: vault1Id,
+      amounts: '1@TSLA'
+    })
+    await testing.container.generate(1)
+
+    { // After loan taken
+      const data = await testing.rpc.loan.getLoanInfo()
+      expect(data).toStrictEqual({
+        ...extendedStartingData,
+        totals: {
+          ...extendedStartingData.totals,
+          loanValue: expect.any(BigNumber)
+        }
+      })
+      // loan value will include some interest
+      expect(data.totals.loanValue.gt(15)).toBeTruthy() // 3 * 5
+      expect(data.totals.loanValue.lt(15.0001)).toBeTruthy()
+    }
+  })
+
+  it('should return with updated loan related governance variables (delayed effect, currentpriceblock must not be affected)', async () => {
+    { // before
+      const data = await testing.rpc.loan.getLoanInfo()
+      expect(data).toStrictEqual(startingData)
+    }
+
+    await testing.rpc.masternode.setGov({
```

Comment:
interesting. I thought `setGov` RPC activates the gov variable at the next block and the `setgovheight` for future heights.

Reply:
Yes, this is to differentiate the with and without height difference.